src/schema-builder.ts

```diff
@@ -1,52 +1,71 @@
         case 'boolean':
-            return { type: ValueType.Boolean };
+            return description ? { type: ValueType.Boolean, description } : { type: ValueType.Boolean };
         case 'string':
-            return { type: ValueType.String };
+            return description ? { type: ValueType.String, description } : { type: ValueType.String };
         case 'object':
             if (value === null) {
-                return { type: ValueType.Null };
+                return description ? { type: ValueType.Null, description } : { type: ValueType.Null };
             }
             if (Array.isArray(value)) {
-                return createSchemaForArray(value, options);
+                return createSchemaForArray(value, options, description);
             }
-            return createSchemaForObject(value, options);
+            return createSchemaForObject(value, options, description);
     }
 }
 
-function createSchemaForArray(arr: Array<any>, options?: SchemaGenOptions): Schema {
+function createSchemaForArray(arr: Array<any>, options?: SchemaGenOptions, description? : string): Schema {
     if (arr.length === 0) {
-        return { type: ValueType.Array };
+        return description ? { type: ValueType.Array, description } : { type: ValueType.Array };
     }
     const elementSchemas = arr.map((value) => createSchemaFor(value, options));
     const items = combineSchemas(elementSchemas);
-    return { type: ValueType.Array, items };
+    return description ? { type: ValueType.Array, items, description } : { type: ValueType.Array, items };
 }
 
-function createSchemaForObject(obj: Object, options?: SchemaGenOptions): Schema {
+function createSchemaForObject(obj: Object, options?: SchemaGenOptions, description? : string): Schema {
     const keys = Object.keys(obj);
+    const nonFunctionKeys = keys.filter((key) => typeof obj[key] !== 'function');
     if (keys.length === 0) {
-        return {
+
+        return description ? {
             type: ValueType.Object,
-        };
+            description
+        } : { type: ValueType.Object };
     }
     const properties = Object.entries(obj).reduce((props, [key, val]) => {
-        props[key] = createSchemaFor(val, options);
+        let description : string | undefined = undefined;
+        
+        if(obj["addDescriptionOfProperty"]){
+            description= obj["addDescriptionOfProperty"](key);
+        }
+
+        if(typeof val !== 'function'){
+            props[key] = createSchemaFor(val, options, description);
+        }
         return props;
     }, {});
 
-    const schema: Schema = { type: ValueType.Object, properties };
+    const schema: Schema = description ? { type: ValueType.Object, properties, description } : { type: ValueType.Object, properties };
+
+    let requiredKeys = nonFunctionKeys;
+    
     if (!options?.noRequired) {
-        schema.required = keys;
+        if(obj["excludeFromRequired"]){
```

Comment:
special, reserved function. Use to specify which properties should be excluded from the `required` of specific object schema. terurns array of names of properties that should not be included in `required` array